NetProbe/Model/NetworkMeasurement.mm

```diff
@@ -263,3 +288,146 @@ -(void) run {
+    .on_log([self](uint32_t type, const char *s) {
+        NSString *current = [NSString stringWithFormat:@"%@: %@", [super getDate],
+                             [NSString stringWithUTF8String:s]];
+        NSLog(@"%s", s);
+        if ((type & MK_LOG_JSON) != 0) {
+            NSData *data = [[NSString stringWithUTF8String:s] dataUsingEncoding:NSUTF8StringEncoding];
+            NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+            self.progress = [[json objectForKey:@"progress"] floatValue];
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [[NSNotificationCenter defaultCenter] postNotificationName:@"reloadTable" object:nil];
+            });
+        }
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self writeOrAppend:current];
+        });
+    })
+    .run([self]() {
+        NSLog(@"web_connectivity testEnded");
+        dispatch_async(dispatch_get_main_queue(), ^{
+            self.completed = TRUE;
+            [TestStorage set_completed:self.test_id];
+            [[NSNotificationCenter defaultCenter] postNotificationName:@"refreshTable" object:self];
+        });
+    });
+}
+
+@end
+
+@implementation NdtTest : NetworkMeasurement
+
+-(id) init {
+    self = [super init];
+    self.name = @"ndt_test";
+    return self;
+}
+
+-(void) run {
+    self.test_id = [NSNumber numberWithDouble:[[NSDate date] timeIntervalSince1970]];
+    self.json_file = [NSString stringWithFormat:@"test-%@.json", self.test_id];
+    self.log_file = [NSString stringWithFormat:@"test-%@.log", self.test_id];
+    [TestStorage add_test:self];
+
+    /*
+    NSBundle *bundle = [NSBundle mainBundle];
+    NSString *geoip_asn = [bundle pathForResource:@"GeoIPASNum" ofType:@"dat"];
+    NSString *geoip_country = [bundle pathForResource:@"GeoIP" ofType:@"dat"];
+    NSString *ca_cert = [bundle pathForResource:@"cacert" ofType:@"pem"];
+    
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory,
+                                                         NSUserDomainMask, YES);
+    NSString *docs_dir = [paths objectAtIndex:0];
+    NSString *ofile = [NSString stringWithFormat:@"%@/output.json", docs_dir];
+    */
+    
+    // Note: the emulator does not cope well with receiving
+    // a signal of type SIGPIPE when the debugger is attached
+    // See http://stackoverflow.com/questions/1294436
+    
+    // XXX: MK_LOG_DEBUG2 sends app to deadlock?!
+    
+    /* Note: this experiment is about comparing the performance of one and
+     multiple streams using the same server; this is why the server address
+     has been explicitly provided below */
+    
+    const char *servers[2] = {
+        "neubot.mlab.mlab1.nuq0t.measurement-lab.org",
+        "52.43.197.62",
+    };
+    static int previous_index = 0;
+    const char *server_address = servers[previous_index++];
+    if (previous_index > 1) {
+        previous_index = 0;
+    }
```

Comment:
You can remove the above snippet of code and the comment above as they are relevant in the specific use case of the application that you pulled code from but not in general.